File: storage/db-core/src/main/java/com/coffee/gu/order/OrderLineJpaRepository.java
package com.coffee.gu.order;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import java.util.List;

public interface OrderLineJpaRepository extends JpaRepository<OrderLineEntity, Long> {
    List<OrderLineEntity> findAllByOrderKeyIn(Collection<String> orderKeys);
    List<OrderLineEntity> findByOrderKey(String orderKey);
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/order/OrderLineJpaRepository.kt
package com.coffee.gu.order

import org.springframework.data.jpa.repository.JpaRepository

interface OrderLineJpaRepository : JpaRepository<OrderLineEntity, Long> {
    fun findAllByOrderKeyIn(orderKeys: Collection<String>): List<OrderLineEntity>
    fun findByOrderKey(orderKey: String): List<OrderLineEntity>
}
